packages/sdk/src/Confidence.ts

```diff
@@ -73,17 +93,26 @@ export class Confidence implements EventSender, Trackable {
   }
 
   setContext(context: Context): void {
+    const current = this.getContext();
+    const changedKeys: string[] = [];
     for (const key of Object.keys(context)) {
-      this.updateContextEntry(key, context[key]);
+      if (Value.equal(current[key], context[key])) continue;
```

Comment:
I think (!) we might be able to do reference equals here.

Reply:
Or actually we won't. There's the case where the child set's the same value as the parent, that would trigger a spurious context change. Added a test for it.